labgrid/driver/serialdriver.py

```diff
@@ -51,13 +51,13 @@ def _read(self, size: int=1, timeout: int=0):
         size -- amount of bytes to read, defaults to 1
         """
         reading = max(size, self.serial.in_waiting)
-        self.logger.debug("Reading %s (min %s) bytes with %s timeout",
-                          reading, size, timeout)
         self.serial.timeout = timeout
         res = self.serial.read(reading)
-        self.logger.debug("Read bytes (%s) or timeout reached", res)
-        if not res:
-            raise TIMEOUT("Timeout exceeded")
+        if res:
+            self.logger.debug("Read %i bytes: %s, timeout %.2f, min=%i",
+                              len(res), res, timeout, size)
```

Comment:
Maybe you could interchange the `timeout` and `min` fields here. Otherwise one could assume that `min` refers to a minimum timeout instead of the minimum bytes to read.